File: Cpp/83.cpp
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode *prev=head;
        ListNode *tmp=prev;
        while(prev){
            while(tmp&&tmp->val==prev->val){
                tmp=tmp->next;
            }
            prev->next=tmp;
            prev=tmp;
        }
        return head;
    }
};

File: Python/83.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        prev = cur = head
        while prev:
            while cur and cur.val == prev.val:
                cur = cur.next
            prev.next = cur
            prev = prev.next
        return head
